services/provider/server/server.go

```diff
@@ -189,3 +205,150 @@ func newClient() (client.Client, error) {
+	}
+
+	extR = append(extR, &pb.ExternalResource{Name: monSecret, Kind: "Secret", Data: mustMarshal(map[string]string{
+		"fsid":         fsid,
+		"mon-secret":   "mon-secret",
+		"admin-secret": "admin-secret",
+	})})
+
+	// Service for monitoring endpoints
+	scMonitoring := &v1.Service{}
+	err = s.client.Get(ctx, types.NamespacedName{Name: "rook-ceph-mgr", Namespace: s.namespace}, scMonitoring)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get rook-ceph-mgr service. %v", err)
+	}
+
+	if scMonitoring.Spec.ClusterIP == "" || strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)) == "" {
+		return nil, fmt.Errorf("service rook-ceph-mgr clusterIP or port is empty")
+	}
+
+	extR = append(extR, &pb.ExternalResource{Name: "monitoring-endpoint", Kind: "CephCluster", Data: mustMarshal(map[string]string{
+		"MonitoringEndpoint": scMonitoring.Spec.ClusterIP,
+		"MonitoringPort":     strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)),
+	})})
+
+	for _, i := range consumerResource.Status.CephResources {
+		data := map[string]string{}
+		if i.Kind == "CephClient" && strings.Contains(i.Name, "-health-checker") {
+			cephUserSecret := &v1.Secret{}
+			err = s.client.Get(ctx, types.NamespacedName{Name: i.CephClients["node"], Namespace: s.namespace}, cephUserSecret)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get %s secret. %v", i.CephClients["node"], err)
+			}
+			data["userID"] = i.Name
+			data["userKey"] = string(cephUserSecret.Data[i.Name])
+
+			extR = append(extR, &pb.ExternalResource{
+				Name: i.Name,
+				Kind: "Secret",
+				Data: mustMarshal(data),
+			})
+		} else if i.Kind == "CephBlockPool" {
+			if strings.Contains(i.Name, "-cephfs-") {
+				if i.CephClients["provisioner"] == "" {
+					return nil, fmt.Errorf("csi provisioner secret name is empty inside status")
+				}
+				cephUserSecret := &v1.Secret{}
+				err = s.client.Get(ctx, types.NamespacedName{Name: i.CephClients["provisioner"], Namespace: s.namespace}, cephUserSecret)
+				if err != nil {
+					return nil, fmt.Errorf("failed to get %s secret. %v", i.CephClients["provisioner"], err)
+				}
+
+				data["adminID"] = i.Name
+				data["adminKey"] = string(cephUserSecret.Data[i.Name])
+
+				extR = append(extR, &pb.ExternalResource{
+					Name: i.CephClients["provisioner"],
+					Kind: "Secret",
+					Data: mustMarshal(data),
+				})
+			} else {
+				if i.CephClients["node"] == "" {
+					return nil, fmt.Errorf("csi node secret name is empty inside status")
+				}
+
+				cephUserSecret := &v1.Secret{}
+				err = s.client.Get(ctx, types.NamespacedName{Name: i.CephClients["node"], Namespace: s.namespace}, cephUserSecret)
+				if err != nil {
+					return nil, fmt.Errorf("failed to get %s secret. %v", i.CephClients["node"], err)
+				}
+
+				data["userID"] = i.Name
+				data["userKey"] = string(cephUserSecret.Data[i.Name])
+
+				extR = append(extR, &pb.ExternalResource{
+					Name: i.CephClients["node"],
+					Kind: "Secret",
+					Data: mustMarshal(data),
+				})
+			}
```

Comment:
Why are we mixing the creation of clients and storage classes?
These are different concerns, you don't need to create them here. Let's move the creation of the client secrets to the CephClinet section above